python-threatexchange/threatexchange/signal_type/index.py

```diff
@@ -139,6 +140,12 @@ def __eq__(self, other: t.Any) -> bool:
 IndexMatch = IndexMatchUntyped[SignalSimilarityInfo, T]
 
 
+@dataclass
+class IndexMatchWithRotation(t.Generic[T]):
```

Comment:
See note about putting this just in match command. It can also be private there (_IndexMatchWithRotation)